source/base/timer.cc

```diff
@@ -861,6 +861,190 @@ TimerOutput::print_summary() const
 
 
 
+void
+TimerOutput::print_summary_of_wall_time_statistics(const MPI_Comm mpi_comm,
+                                                   const double quantile) const
+{
+  // we are going to change the precision and width of output below. store the
+  // old values so we can restore it later on
+  const std::istream::fmtflags old_flags = out_stream.get_stream().flags();
+  const std::streamsize old_precision    = out_stream.get_stream().precision();
+  const std::streamsize old_width        = out_stream.get_stream().width();
```

Comment:
What about
```
boost::io::ios_flags_saver ifs(out_stream.get_stream());
```

Reply:
Looks good - I have used this in the new function here and further above in `print_summary()` to remove 12 lines of code :wink: